snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedSnapshotStore.java

```diff
@@ -515,27 +539,37 @@ FileBasedSnapshot newSnapshot(
               currentPersistedSnapshotRef.get()));
     }
 
+    availableSnapshots.add(newPersistedSnapshot);
+
     LOGGER.info("Committed new snapshot {}", newPersistedSnapshot.getId());
 
     snapshotMetrics.incrementSnapshotCount();
     observeSnapshotSize(newPersistedSnapshot);
 
-    LOGGER.trace(
-        "Purging snapshots older than {}",
-        newPersistedSnapshot.getMetadata().getSnapshotIdAsString());
-    if (currentPersistedSnapshot != null) {
-      LOGGER.debug(
-          "Deleting previous snapshot {}",
-          currentPersistedSnapshot.getMetadata().getSnapshotIdAsString());
-      currentPersistedSnapshot.delete();
-    }
-    purgePendingSnapshots(newPersistedSnapshot.getMetadata());
+    deleteOlderSnapshots(newPersistedSnapshot);
 
     listeners.forEach(listener -> listener.onNewSnapshot(newPersistedSnapshot));
 
     return newPersistedSnapshot;
   }
 
+  private void deleteOlderSnapshots(final FileBasedSnapshot newPersistedSnapshot) {
+    LOGGER.trace(
+        "Purging snapshots older than {}",
+        newPersistedSnapshot.getMetadata().getSnapshotIdAsString());
+    final var snapshotsToDelete =
+        availableSnapshots.stream()
+            .filter(s -> !s.getId().equals(newPersistedSnapshot.getId()))
```

Comment:
:wrench: Almost seems a bit silly but could we add a test that a newly taken snapshot is not immediately deleted if it is not reserved?

Reply:
There are other tests that verify that after a snapshot is committed, its files exists. So I did not add a new test, but added extra asserts in the reservation tests to verify that the newSnapshot is not deleted.